### ISSUE TITLE
data race in TestRevision

### ISSUE THREAD

**Issue by xiang90** [coreos/etcd#8182]
```
==================
WARNING: DATA RACE
Write at 0x00c4201e5250 by goroutine 47:
  github.com/coreos/etcd/compactor.TestRevision()
      /home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/compactor/revision_test.go:45 +0x3da
  testing.tRunner()
      /home/travis/.gimme/versions/go1.8.3.linux.amd64/src/testing/testing.go:657 +0x107
Previous write at 0x00c4201e5250 by goroutine 48:
  github.com/coreos/etcd/compactor.(*fakeRevGetter).Rev()
      /home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/compactor/compactor_test.go:39 +0xb9
  github.com/coreos/etcd/compactor.(*Revision).Run.func1()
      github.com/coreos/etcd/compactor/_test/_obj_test/revision.go:79 +0x26f
Goroutine 47 (running) created at:
  testing.(*T).Run()
      /home/travis/.gimme/versions/go1.8.3.linux.amd64/src/testing/testing.go:697 +0x543
  testing.runTests.func1()
      /home/travis/.gimme/versions/go1.8.3.linux.amd64/src/testing/testing.go:882 +0xaa
  testing.tRunner()
      /home/travis/.gimme/versions/go1.8.3.linux.amd64/src/testing/testing.go:657 +0x107
  testing.runTests()
      /home/travis/.gimme/versions/go1.8.3.linux.amd64/src/testing/testing.go:888 +0x4e0
  testing.(*M).Run()
      /home/travis/.gimme/versions/go1.8.3.linux.amd64/src/testing/testing.go:822 +0x1c3
  main.main()
      github.com/coreos/etcd/compactor/_test/_testmain.go:100 +0x33d
Goroutine 48 (running) created at:
  github.com/coreos/etcd/compactor.(*Revision).Run()
      github.com/coreos/etcd/compactor/_test/_obj_test/revision.go:99 +0x1af
  github.com/coreos/etcd/compactor.TestRevision()
      /home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/compactor/revision_test.go:38 +0x349
  testing.tRunner()
      /home/travis/.gimme/versions/go1.8.3.linux.amd64/src/testing/testing.go:657 +0x107
==================
2017-06-27 20:08:29.624276 N | compactor: Starting auto-compaction at revision 90 (retention: 10 revisions)
2017-06-27 20:08:29.634612 N | compactor: Finished auto-compaction at revision 90
2017-06-27 20:08:34.645146 N | compactor: Starting auto-compaction at revision 190 (retention: 10 revisions)
2017-06-27 20:08:34.655853 N | compactor: Finished auto-compaction at revision 190
--- FAIL: TestRevision (5.05s)
	testing.go:610: race detected during execution of test
2017-06-27 20:08:34.677084 N | compactor: Starting auto-compaction at revision 90 (retention: 10 revisions)
2017-06-27 20:08:34.687712 N | compactor: Finished auto-compaction at revision 90
FAIL
```

**Comment by yudai** [comment]
`rg.Rev` in the test should be protected by a mutex. I'll send a PR to fix.

**Comment by xiang90** [comment]
fixed by #8133
